core/tests/ts-integration/tests/custom-account.test.ts

```diff
@@ -95,6 +113,94 @@ describe('Tests for the custom account behavior', () => {
+        await expect(
+            sendCustomAccountTransaction(
+                tx as zksync.types.Transaction,
+                alice.provider,
+                await customAccount.getAddress(),
+                testMaster.environment().l2ChainId
+            )
+        ).toBeRejected(
+            'failed to validate the transaction. reason: Violated validation rules: block.timestamp range is too short'
+        );
+    });
+
+    test('Should fail transaction validation due to timestamp assertion in the validation tracer - close to the range end', async () => {
+        const now = Math.floor(Date.now() / 1000);
+        const rangeStart = now - 600;
+        const rangeEnd = now + 30;
+
+        const customAccount = await deployAndFundCustomAccount(
+            alice,
+            erc20Address,
+            timestampAsserterAddress,
+            rangeStart,
+            rangeEnd
+        );
+
+        const tx = await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT);
+
+        await expect(
+            sendCustomAccountTransaction(
+                tx as zksync.types.Transaction,
+                alice.provider,
+                await customAccount.getAddress(),
+                testMaster.environment().l2ChainId
+            )
+        ).toBeRejected(
+            'failed to validate the transaction. reason: Violated validation rules: block.timestamp is too close to the range end'
+        );
+    });
+
+    test('Should fail to estimate fee due to block.timestamp assertion in the smart contract', async () => {
+        const now = Math.floor(Date.now() / 1000);
+        const rangeStart = now + 300;
+        const rangeEnd = now + 1000;
+
+        const customAccount = await deployAndFundCustomAccount(
+            alice,
+            erc20Address,
+            timestampAsserterAddress,
+            rangeStart,
+            rangeEnd
+        );
+
+        const promise = alice.provider.estimateGas({
+            ...(await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT)),
+            from: await customAccount.getAddress()
+        });
+
+        // requires custom assertion because the transaction would fail in "estimateFee" that is not handled correctly by zksync ethers
+        promise
+            .then((_) => {
+                fail('The transaction was expected to fail due to timestamp assertion');
+            })
+            .catch((error) => {
+                // If it's an error object with the structure you're mentioning, check its properties
+                if (error.info && error.info.error && error.info.error.message) {
+                    expect(error.info.error.message).toEqual(
+                        'failed to validate the transaction. reason: Timestamp is out of range'
+                    );
+                } else {
+                    fail('The transaction was expected to fail with a different message');
```

Comment:
Please use `expect(null).fail(..)` as in other integration tests.

Reply:
Fixed